src/components/state-lookup.ts

```diff
@@ -146,13 +145,20 @@ export class StateLookup {
      * cannot be tracked.
      */
     public trackRoom(roomId: string) {
-        const r = this.dict[roomId] = this.dict[roomId] || {};
+        const r = this.dict[roomId] = this.dict[roomId] || {
+            syncPending: false,
+        };
         if (r.syncPromise) {
             return r.syncPromise;
         }
         r.events = {};
-        // For backwards compat, we use Bluebird
-        r.syncPromise = Bluebird.resolve(this.getInitialState(roomId));
+
+        r.syncPromise = (async () => {
+            r.syncPending = true;
+            const res = await this.getInitialState(roomId);
+            r.syncPending = false;
+            return res;
+        })();
 
         return r.syncPromise;
```

Comment:
We may want to ensure the state is set to pending before the promise gets run as we already know its pending.
```suggestion
        r.syncPending = true;
        r.syncPromise = (async () => {
            const res = await this.getInitialState(roomId);
            r.syncPending = false;
            return res;
        })();

        return r.syncPromise;
```